components/webdriver_server/lib.rs

```diff
@@ -840,6 +857,49 @@ impl Handler {
         result
     }
 
+    fn wait_for_navigation(&self) -> WebDriverResult<WebDriverResponse> {
+        let browsing_context_id = self.session()?.browsing_context_id;
+
+        match self.load_status_receiver.try_recv() {
+            Ok(WebDriverLoadStatus::NavigationStarted) => {
+                // Check if the navigation is approved or not with constellation
+                match self.load_status_receiver.recv() {
+                    Ok(WebDriverLoadStatus::Loading) => {
+                        self.wait_for_document_ready_state()?;
+                        Ok(WebDriverResponse::Void)
+                    },
+                    Ok(WebDriverLoadStatus::Canceled) => Ok(WebDriverResponse::Void),
+                    Ok(_) => unreachable!("Unexpected load status received"),
+                    Err(_) => Ok(WebDriverResponse::Void),
+                }
+            },
+            Ok(WebDriverLoadStatus::NavigationHashChanged) => {
```

Comment:
```suggestion
            Ok(WebDriverLoadStatus::NavigateToFragment) => {
```
Let's make the naming consistent with the spec and other parts of the code: https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-the-fragment-identifier